cmd/attach/policy/attach_policy_service.go

```diff
@@ -0,0 +1,153 @@
+// verify role has quota to attach the policies
+func (p *policyService) ValidateOptions(roleName string, policyArns []string) error {
+	role, err := p.AWSClient.GetRoleByName(roleName)
+	if err != nil {
+		return fmt.Errorf(
+			"Failed to find the role %s: %s",
+			roleName, err,
+		)
+	}
+	isRedHatManaged := false
+	for _, tag := range role.Tags {
+		if awsutil.ToString(tag.Key) == tags.RedHatManaged &&
+			awsutil.ToString(tag.Value) == "true" {
+			isRedHatManaged = true
+			break
+		}
+	}
+	if !isRedHatManaged {
+		return fmt.Errorf("Rosa only supports attaching policy to RedHat managed role")
+	}
+
+	err = validatePolicyQuota(p.AWSClient, roleName, policyArns)
+	if err != nil {
+		return err
+	}
+	for _, policyArn := range policyArns {
+		fmt.Printf("validate policy %s\n", policyArn)
+		_, err = p.AWSClient.IsPolicyExists(policyArn)
+		if err != nil {
+			return fmt.Errorf(
+				"Failed to find the policy %s: %s",
+				policyArn, err,
+			)
+		}
+	}
+	return nil
+}
+
+func (p *policyService) AutoAttachArbitraryPolicy(roleName string, policyArns []string,
+	accountID, orgID string) (string, error) {
+	output := ""
+	for _, policyArn := range policyArns {
+		err := p.AWSClient.AttachRolePolicy(roleName, policyArn)
+		if err != nil {
+			return output, fmt.Errorf("Failed to attach policy %s to role %s", policyArn, roleName)
+		}
+		output = output + fmt.Sprintf("Attached policy '%s' to role '%s'\n", policyArn, roleName)
+		p.OCMClient.LogEvent("ROSAAttachPolicyAuto", map[string]string{
+			ocm.Account:      accountID,
+			ocm.Organization: orgID,
+			ocm.RoleName:     roleName,
+			ocm.PolicyArn:    policyArn,
+		})
+	}
+	return output, nil
+}
+
+func (p *policyService) ManualAttachArbitraryPolicy(roleName string, policyArns []string,
+	accountID, orgID string) string {
+	cmd := ""
+	for _, policyArn := range policyArns {
+		cmd = cmd + fmt.Sprintf("aws iam attach-role-policy --role-name %s --policy-arn %s\n",
+			roleName, policyArn)
+		p.OCMClient.LogEvent("ROSAAttachPolicyManual", map[string]string{
+			ocm.Account:      accountID,
+			ocm.Organization: orgID,
+			ocm.RoleName:     roleName,
+			ocm.PolicyArn:    policyArn,
+		})
+	}
+	return cmd
+}
+
+func validatePolicyQuota(c aws.Client, roleName string, policyArns []string) error {
+	quota, err := c.GetIAMServiceQuota(quotaCode)
+	if err != nil {
+		return fmt.Errorf("Failed to get quota of policies per role: %s", err)
+	}
+	attachedPolicies, err := c.GetAttachedPolicy(&roleName)
+	if err != nil {
+		return fmt.Errorf("Failed to get attached policies of role %s: %s", roleName, err)
+	}
+	if len(policyArns)+len(attachedPolicies) > int(*quota.Quota.Value) {
+		policySkipped := 0
+		for _, attachedPolicy := range attachedPolicies {
+			if slices.Contains(policyArns, attachedPolicy.PolicyArn) {
+				policySkipped++
+			}
+		}
+		if len(policyArns)+len(attachedPolicies)-policySkipped > int(*quota.Quota.Value) {
+			return fmt.Errorf("Exceeds the quota of policies per role: %d."+
```

Comment:
I was wondering if a more concise error message here would be easier to read:

"Failed to attach policies due to quota limitations (total limit: x, expected: y)."
x = quota value
y = total of "len(policyArns)+len(attachedPolicies)-policySkipped)"

Reply:
message updated